### PR TITLE
Fix autocomplete input debounce

### PR DESCRIPTION
**Pull request by lab900-winand-vandenbergh**

### CODE REVIEW

File: projects/forms/src/lib/components/form-fields/autocomplete-field/autocomplete-field.component.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { AfterViewInit, Component, ElementRef, HostBinding, ViewChild } from '@angular/core';
 import { FormComponent } from '../../../models/IFormComponent';
 import { AutocompleteOptions, ValueLabel } from '../../../models/FormField';
-import { Observable, isObservable, of, fromEvent } from 'rxjs';
+import { Observable, isObservable, of, fromEvent, Subject, BehaviorSubject } from 'rxjs';
```
Comment (lab900-senn-lambrecht) [inline projects/forms/src/lib/components/form-fields/autocomplete-field/autocomplete-field.component.ts]: subject unused?

File: projects/forms/src/lib/components/form-fields/autocomplete-field/autocomplete-field.component.ts [hunk 2]
```diff
@@ -29,13 +31,16 @@ export class AutocompleteFieldComponent extends FormComponent<AutocompleteOption
     this.initFilteredOptionsListener();
   }
 
+  public inputChanged($event: Event): void {
+    this.inputChange.next(($event.target as any).value);
```
Comment (lab900-senn-lambrecht) [inline projects/forms/src/lib/components/form-fields/autocomplete-field/autocomplete-field.component.ts:33]: as string

File: projects/forms/src/lib/components/form-fields/autocomplete-multiple-field/autocomplete-multiple-field.component.ts [hunk 3]
```diff
@@ -30,9 +33,24 @@ export class AutocompleteMultipleFieldComponent extends FormComponent<Autocomple
...
 
+  public ngAfterViewInit(): void {
+    super.ngAfterViewInit();
+    this.initFilteredOptionsListener();
+  }
+
   public inputChanged($event: Event): void {
-    const res = this.options.autocompleteOptions(($event.target as any).value, this.fieldControl);
-    this.filteredOptions = isObservable(res) ? res : of(res);
+    this.inputChange.next(($event.target as any).value);
```
Comment (lab900-senn-lambrecht) [inline projects/forms/src/lib/components/form-fields/autocomplete-multiple-field/autocomplete-multiple-field.component.ts:42]: as string

File: projects/forms/src/lib/components/form-fields/autocomplete-field/autocomplete-field.component.html [hunk 4]
```diff
@@ -2,6 +2,7 @@
     <mat-label *ngIf="schema.title" translate>{{ schema.title }}</mat-label>
     <input
         #input
+        (input)="this.inputChanged($event)"
```
Comment (lab900-senn-lambrecht) [inline projects/forms/src/lib/components/form-fields/autocomplete-field/autocomplete-field.component.html]: remove this.

**Review by lab900-senn-lambrecht — CHANGES_REQUESTED** [review]

File: projects/forms/src/lib/components/form-fields/autocomplete-multiple-field/autocomplete-multiple-field.component.ts (hunk 3 above)
Comment (lab900-winand-vandenbergh) [reply projects/forms/src/lib/components/form-fields/autocomplete-multiple-field/autocomplete-multiple-field.component.ts:42]: it's the value that's a string, not the target. I'm not sure how to type the target. I think this any typing is code you wrote as well :p

File: projects/forms/src/lib/components/form-fields/autocomplete-field/autocomplete-field.component.html (hunk 4 above)
Comment (lab900-winand-vandenbergh) [reply projects/forms/src/lib/components/form-fields/autocomplete-field/autocomplete-field.component.html]: why? this is being used to set the subject.

Comment (lab900-senn-lambrecht) [reply projects/forms/src/lib/components/form-fields/autocomplete-field/autocomplete-field.component.html]: I meant just the "this." the this context isn't needed.

**Review by lab900-senn-lambrecht — APPROVED** [review]

File: projects/forms/src/lib/components/form-fields/autocomplete-field/autocomplete-field.component.ts (hunk 2 above)
Comment (lab900-winand-vandenbergh) [reply projects/forms/src/lib/components/form-fields/autocomplete-field/autocomplete-field.component.ts:33]: only the value of the $event.target is a string. I don't know how the $event.target should be typed. This was already typed as any by you.

**Pull request merged**